system/lib/pthread/emscripten_futex_wait.c

```diff
@@ -102,7 +102,7 @@ static int futex_wait_main_browser_thread(volatile void* addr,
     // here and return, it's the same is if what happened on the main thread
     // was the same as calling _emscripten_yield()
     // a few times before calling emscripten_futex_wait().
-    if (__c11_atomic_load((_Atomic uintptr_t*)addr, __ATOMIC_SEQ_CST) != val) {
+    if (__c11_atomic_load((_Atomic uint32_t*)addr, __ATOMIC_SEQ_CST) != val) {
```

Comment:
This feels wrong but it's the only way I could get it working. I've tried changing argument type to uintptr_t but it fails with "i64.atomic.load: this operation does not support unaligned access". Most likely it needs to be fixed somewhere else

Reply:
Since `val` is uint32_t I think this actually makes sense.    I don't think we ever want this to compile to an i64 operation.

We might want to change the first argument to `emscripten_futex_wait` to be `uint32_t*` too.